python/ql/src/experimental/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -130,7 +130,7 @@ module EssaFlow {
   }
 
   predicate useToNextUse(NameNode nodeFrom, NameNode nodeTo) {
-    AdjacentUses::adjacentUseUseSameVar(nodeFrom, nodeTo)
+    AdjacentUses::adjacentUseUse(nodeFrom, nodeTo)
```

Comment:
I guess this change was motivated to be able to handle this new part of the test, but I wasn't 100% sure.

```diff
     if emulated_is_safe(s):
         ensure_not_tainted(s)
+        s = TAINTED_STRING
+        ensure_tainted(s)
```

Reply:
Ah, this was a bit unstructured. Basically, I am going back on my previous opinion since I noticed that `firstUse` also tracks through phi-nodes. So I went ahead and added `adjacentUseUse`. I think we should probably remove phi- (and pi-)nodes from the data flow graph later. It is slightly cheeky to do that change in this PR..

Reply:
Please add this to our internal tracking issue so we remember to look into this :wink:

Reply:
Done